torch_geometric/graphgym/logger.py

```diff
@@ -247,3 +260,105 @@ def create_logger():
     for i, dataset in enumerate(range(cfg.share.num_splits)):
         loggers.append(Logger(name=names[i], task_type=infer_task()))
     return loggers
+
+
+class LoggerCallback(Callback):
+    def __init__(self):
+        self._logger = create_logger()
+        self._train_epoch_start_time = None
+        self._val_epoch_start_time = None
+        self._test_epoch_start_time = None
+
+    def _create_stats(self, time_start, outputs: Dict[str, Any],
+                      trainer: "pl.Trainer") -> Dict:
+        true: torch.Tensor = outputs["true"]
+        pred: torch.Tensor = outputs["pred"]
+        pred_score: torch.Tensor = outputs["pred_score"]
+        loss: torch.Tensor = outputs["loss"]
+        lr = trainer.lr_scheduler_configs[0].scheduler.get_last_lr()[0]
+        stats = dict(true=true.detach().cpu(), pred=pred.detach().cpu(),
```

Comment:
Any chance we can test this logger in this PR already?

Reply:
The Logger is a PL callback, I can mock training and test it out in this PR.

Reply:
I have added a training test with logger as callback [here](https://github.com/aniketmaurya/pytorch_geometric/blob/implement_module/test/graphgym/test_graphgym.py#L195)

This would only test if the hooks get any error or not. Anything specific you would like to check here @rusty1s ?

Suggestions @Borda?

Reply:
I have added a test for logger callback [here](https://github.com/aniketmaurya/pytorch_geometric/blob/implement_module/test/graphgym/test_graphgym.py#L195)

This would only test if the callback hooks get any error or not. Anything specific you would like to check here @rusty1s ?

Suggestions @Borda?